codegen/golang.go

```diff
@@ -139,7 +146,17 @@ func (t *TableDefinition) addColumnFromField(field reflect.StructField, parent *
 		return
 	}
 
-	columnType, err := valueToSchemaType(field.Type)
+	var overrideType *schema.ValueType
+	if t.valueTypeOverride != nil {
+		overrideType = t.valueTypeOverride(field)
```

Comment:
I think the type definition for `valueTypeOverride` should be the same as `valueToSchemaType` i.e also return an error so we propogate this, log it, have consistent behaviour and so on.

Reply:
Maybe remove the default value fallback. If there's an override, always use that. The override can decide to call `ValueToSchemaType` on its own if it doesn't want any changes (we need to export that fn...)

Reply:
Yeah agree, also no need for a fallback. i.e if there is an error then it just errors out. (returning nil will panic but that's fine as this is undefined behaviour if no error is returned)

Reply:
Do we want to expose `valueToSchemaType`? It's private at the moment

Reply:
if we remove the fallback and expose the inner func then no need for it to return a ptr, it can have the exact same signature.

Reply:
> Yeah agree, also no need for a fallback. i.e if there is an error then it just errors out. (returning nil will panic but that's fine as this is undefined behaviour if no error is returned)

Should we expose `valueToSchemaType` as a package function or a struct function for the table?
I'm not sure that looks nicer that handling the fallback ourselves